Jenkinsfile

```diff
@@ -0,0 +1,247 @@
+        disableConcurrentBuilds()
+        timestamps()
+    }
+    triggers {
+        /*
+          Restrict nightly builds to master branch, all others will be built on change only.
+          Note: The BRANCH_NAME will only work with a multi-branch job using the github-branch-source
+        */
+        cron(env.BRANCH_NAME == "master" ? "H H(21-23) * * *" : "")
+    }
+    stages {
+        stage('Setup') {
+            steps {
+                script {
+                    slackSend color: 'good', message: "STARTED: ${JOB_NAME} ${BUILD_NUMBER} ${BUILD_URL}"
+                    withCredentials([usernameColonPassword(credentialsId: 'cxbot', variable: 'GITHUB_TOKEN')]) {
+                        postCommentIfPR("Internal build has been started. Your results will be available at completion. See build progress in [legacy Jenkins UI](${BUILD_URL}) or in [Blue Ocean UI](${BUILD_URL}display/redirect).", "${GITHUB_USERNAME}", "${GITHUB_REPONAME}", "${GITHUB_TOKEN}")
+                    }
+                    if (params.RELEASE == true) {
+                        if (params.RELEASE_VERSION != 'NA') {
+                            env.RELEASE_VERSION = params.RELEASE_VERSION
+                        } else {
+                            echo ("Setting release version to ${getBaseVersion()}")
+                            env.RELEASE_VERSION = getBaseVersion()
+                        }
+                        if (params.RELEASE_TAG != 'NA') {
+                            env.RELEASE_TAG = params.RELEASE_TAG
+                        } else {
+                            echo("Setting release tag")
+                            env.RELEASE_TAG = "${env.GITHUB_REPONAME}-${env.RELEASE_VERSION}"
+                        }
+                        if (params.NEXT_VERSION != 'NA') {
+                            env.NEXT_VERSION = params.NEXT_VERSION
+                        } else {
+                            echo("Setting next version")
+                            env.NEXT_VERSION = getDevelopmentVersion()
+                        }
+                        echo("Release parameters: release-version: ${env.RELEASE_VERSION} release-tag: ${env.RELEASE_TAG} next-version: ${env.NEXT_VERSION}")
+                    }
+                }
+            }
+        }
+        // The incremental build will be triggered only for PRs. It will build the differences between the PR and the target branch
+        stage('Incremental Build') {
+            when {
+                allOf {
+                    expression { env.CHANGE_ID != null }
+                    expression { env.CHANGE_TARGET != null }
+                }
+            }
+            parallel {
+                stage ('Linux') {
+                    steps {
+                        withMaven(maven: 'Maven 3.5.4', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LINUX_MVN_RANDOM}') {
```

Comment:
✏️ Should probably be changed to Java11 somehow. True for the rest of the file

Reply:
We added `jdk11` to Jenkins, so we should just be able to change it in the Jenkinsfile.